### DISCUSSION TITLE
Network socket error when trying to connect to REST data source

### DISCUSSION THREAD

**Discussion by pjfrey** [Budibase/budibase#6430] [Help]
I'm not sure if this is me doing something wrong or on Budibase's side. I've connected a REST data source, provided the respective auth and header as well as a body, and when running the query I get the following response:

Query Error: request to [endpoint] failed, reason: Client network socket disconnected before secure TLS connection was established

**Comment by melohagan** [comment]
Hey @purplepatrick 

Just to verify, are you able to successfully connect to the API via Postman or some other external method?

**Reply by pjfrey** [reply]
Yes, daily (been using it via Insomnia for a year)

**Reply by melohagan** [reply]
Is the REST data source a public API, or is it hosted within docker on your machine?

**Reply by pjfrey** [reply]
it's a public API

**Reply by melohagan** [reply]
Where and how is Budibase hosted?

**Reply by pjfrey** [reply]
self-hosted. PostgreSQL on Digital Ocean. However, for this app, I have not set up the db yet and started with the REST source

**Reply by pjfrey** [reply]
sorry, It appears it's cloud hosted (it wasn't set up by me), as I still get the suggestion to self-host in account.

**Reply by melohagan** [reply]
Hmm, there's another issue here around DigitalOcean I wasn't able to resolve: https://github.com/Budibase/budibase/discussions/5512

I'm not great with SSL and certificates, I'll have to get back to you on this one.

**Reply by pjfrey** [reply]
thanks

**Comment by melohagan** [comment]
Hi @purplepatrick 

Did you ever solve this or find a workaround?

I've raised an issue here to allow REST queries to be made without SSL: https://github.com/Budibase/budibase/issues/6885

**Comment by pjfrey** [comment]
No, I haven't tried it anymore. Is there any indication that it was fixed? I can try again...
